test/e2e/device_get_test.go

```diff
@@ -0,0 +1,214 @@
+package e2e
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+
+	root "github.com/equinix/metal-cli/internal/cli"
+	"github.com/equinix/metal-cli/internal/devices"
+	outputPkg "github.com/equinix/metal-cli/internal/outputs"
+	"github.com/equinix/metal-cli/internal/projects"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCli_Devices_get(t *testing.T) {
+	var projectId, deviceId string
+	subCommand := "device"
+	consumerToken := ""
+	apiURL := ""
+	Version := "metal"
+	rootClient := root.NewClient(consumerToken, apiURL, Version)
+	type fields struct {
+		MainCmd  *cobra.Command
+		Outputer outputPkg.Outputer
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		want    *cobra.Command
+		cmdFunc func(*testing.T, *cobra.Command)
+	}{
+		{
+			name: "create_project",
+			fields: fields{
+				MainCmd:  projects.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				project_name := "metal-cli-test-project"
+				root := c.Root()
+				root.SetArgs([]string{"project", "create", "-n", project_name})
+				rescueStdout := os.Stdout
+				r, w, _ := os.Pipe()
+				os.Stdout = w
+				if err := root.Execute(); err != nil {
+					t.Error(err)
+				}
+				w.Close()
+				out, _ := io.ReadAll(r)
+				os.Stdout = rescueStdout
+				fmt.Print(string(out[:]))
+				if !strings.Contains(string(out[:]), "metal-cli-test-project") {
+					t.Error("expected output should include metal-cli-test-device string in the out string ")
+				}
+
+				// Regular expression pattern to match the ID hash when NAME is "cli-acc-org-test3"
+				idNamePattern := `(?m)^\| ([a-zA-Z0-9-]+) +\| *` + project_name + ` *\|`
+
+				// Find the match of the ID and NAME pattern in the table string
+				match := regexp.MustCompile(idNamePattern).FindStringSubmatch(string(out[:]))
+
+				// Extract the ID from the match
+				if len(match) > 1 {
+					projectId = strings.TrimSpace(match[1])
+				} else {
+					fmt.Println("No projecct ID found for the given NAME.")
+				}
+			},
+		},
+		{
+			name: "create_device",
+			fields: fields{
+				MainCmd:  devices.NewClient(rootClient, outputPkg.Outputer(&outputPkg.Standard{})).NewCommand(),
+				Outputer: outputPkg.Outputer(&outputPkg.Standard{}),
+			},
+			want: &cobra.Command{},
+			cmdFunc: func(t *testing.T, c *cobra.Command) {
+				root := c.Root()
+				root.SetArgs([]string{subCommand, "create", "-p", string(projectId), "-P", "c3.small.x86", "-m", "da", "-O", "ubuntu_20_04", "-H", "metal-cli-test-project-dev"})
```

Comment:
Since `metal device create` has its own test already, you could create the device in this test by using `metal-go` directly.  You would need a state check loop here as well; that could be implemented in a test helpers module that is reused across multiple tests.